src/BizHawk.Client.Common/tools/RamSearchEngine/RamSearchEngine.cs

```diff
@@ -301,28 +307,19 @@ private IEnumerable<long> AllAddresses()
 		{
 			foreach (var watch in _watchList)
 			{
-				if (_settings.CheckMisAligned)
+				var addr = watch.Address;
+				yield return addr;
+				if (_settings.CheckMisAligned) continue; // I guess we're assuming the other parts of the address are also in `_watchList`? --yoshi
```

Comment:
Right... if you have a better idea for how to implement this in a simple way without getting duplicates feel free to do that here.